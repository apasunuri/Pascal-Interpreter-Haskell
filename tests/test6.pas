program test6;

var 
    a, b : real;

begin
    a := 0;
    b := 6;
    var
        b : real;
    while (a <= 9) do
    begin
        b := ln(a);
        case(sqrt(a)) of
            1 : begin
                    writeln('a is a perfect square');
                    writeln('natural log of a');
                    writeln(b);
                    writeln();
                end;
            2 : begin
                    writeln('a is a perfect square');
                    writeln('natural log of a');
                    writeln(b);
                    writeln();
                end;
            3 : begin
                    writeln('a is a perfect square');
                    writeln('natural log of a');
                    writeln(b);
                    writeln();
                end;
        end;
        a := a + 1;
    end;
    a := 4;
    writeln(a);
    writeln(b); 
end.