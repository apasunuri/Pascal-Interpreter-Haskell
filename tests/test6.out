a is a perfect square
natural log of a
0.0

a is a perfect square
natural log of a
1.3862944

a is a perfect square
natural log of a
2.1972246

4.0
6.0